### DISCUSSION TITLE
I've installed rustdesk from deb file (linux mint 22 cinnamon) and it "hijacked" my system (daemon / background service auto start)

### DISCUSSION THREAD

**Discussion by tethragon** [rustdesk/rustdesk#9284] [General]
### Bug Description

When I first run rustdek, I think a service started that there was no way to turn it off. Quitting the program was not possible. Even killing the rustdesk process from system monitor was not possible (the process was starting again). Also, after reboot the process was there and rustdesk was running in the background.

This is unacceptable. Rustdesk program (and all relative processes) should be terminated when I quit the program.

### How to Reproduce

Install from deb and run the program. Then try to quit. It won't be possible.

### Expected Behavior

Quitting a program should quit the program and all processes relative to it.

### Operating system(s) on local side and remote side

Linux mint (didn't conect to remote side

### RustDesk Version(s) on local side and remote side

1.3.0 (didn't connect to remote side

### Screenshots

No screenshot available. The description is enough.

**Comment by rustdesk** [comment]
Most users require this #3571 #7100 for unattended access, but you call this is "hijacked". How can we make an APP everyone like? :(

If you do not like this, there are many ways, any one of below.

- Turn this one
![image](https://github.com/user-attachments/assets/2bfd1309-1bea-49d6-b8ef-d9dc945c79af)

- Uninstall deb version, use the Flatpak or AppImage version.

- Stop service or run `sudo systemctl stop rustdesk; sudo systemctl disable rustdesk` (this has a problem if you reinstall/update rustdesk).
![image](https://github.com/user-attachments/assets/47f62dd5-e642-4e82-a054-7c5306ccfadd)

- Exit from tray, but there is an issue <= 1.3.0, https://github.com/rustdesk/rustdesk/issues/9286

**Reply by tethragon** [reply]
Stopping the service didn't solve the problem as it was still impossible to quit the app. If this is what users want I am sorry for not understanding it. Is it ok not being able to quit the app? I understand the need for unattended use case but there should be an option in the app to be able to QUIT. Thank you for your answer. I have already uninstalled rustdesk and use appimage version which actually quits when you want to. Thank you for your great app btw

**Reply by Sunil178** [reply]
> Most users require this #3571 #7100 for unattended access, but you call this is "hijacked". How can we make an APP everyone like? :(
> 
> If you do not like this, there are many ways, any one of below.
> 
> * Turn this one
>   ![image](https://private-user-images.githubusercontent.com/71636191/365357749-2bfd1309-1bea-49d6-b8ef-d9dc945c79af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTYxNjgsIm5iZiI6MTcyNTcxNTg2OCwicGF0aCI6Ii83MTYzNjE5MS8zNjUzNTc3NDktMmJmZDEzMDktMWJlYS00OWQ2LWI4ZWYtZDlkYzk0NWM3OWFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDEzMzEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMjBkZmM0ZjQ5NjBiMGFhMGFhOWVlMDkxMTc1YjkxMmM2ODI1MDBiNTE3ZDliYjZhMWQ2ZmY3YTdhZmYxNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HgRsq2nW6EohzLHJ3iB14WucUbip7SV3h9GhLb4Mfa8)
> * Uninstall deb version, use the Flatpak or AppImage version.
> * Stop service or run `sudo systemctl stop rustdesk; sudo systemctl disable rustdesk` (this has a problem if you reinstall/update rustdesk).
>   ![image](https://private-user-images.githubusercontent.com/71636191/365357787-47f62dd5-e642-4e82-a054-7c5306ccfadd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTYxNjgsIm5iZiI6MTcyNTcxNTg2OCwicGF0aCI6Ii83MTYzNjE5MS8zNjUzNTc3ODctNDdmNjJkZDUtZTY0Mi00ZTgyLWEwNTQtN2M1MzA2Y2NmYWRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDEzMzEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkMzMwMzlhNjVkZjc1OGQ0YmZhMWQ4ZjFkODMyOGI5Nzc0MmVlMmM3YjFjZDEzODIzMzU2MjAxOTczMjg2NDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IdffbAttbGzqWKjTOj7-PcCYmi87ysaLUAMvjyEg68o)

I tried the same, but didn't help

**Comment by marc-silva** [comment]
ChatGPT found the answer for me in Method 1 that it provided:
https://chatgpt.com/share/66eaf638-c2dc-800c-b517-2a86904500da

Basically, it is not in any standard autostart folder. It was added to systemd, likely so that it starts earlier in the boot process before anyone even logs in. This is likely better because, the computer can just be turned on and Rustdesk is already running and allowing connections. It does not need to have have someone login locally to get the application running from autostart.

Because of this, perhaps it is harder to shutdown and cannot be shutdown from the app itself. Maybe because you cannot specify super user rights when trying to quit or exit it in the tray icon or from the application itself. You need to do it with superuser rights (sudo) from the terminal:

Confirm it is in systemd: 
`systemctl list-units --type=service | grep rustdesk`

If there, disable it:
`sudo systemctl disable rustdesk.service`

Stop the sevice now: 
`sudo systemctl stop rustdesk.service`

Rustdesk is a great program and I am greatful they have Linux support. Perhaps in a future release, they can find a way to allow users to control (turn off) its autostart function in systemd as well as exit/stop the service through the application task tray icon and GUI.

_Note that this method worked for me using Lubuntu 24.04.1 and installing **rustdesk-1.3.0-x86_64.deb** from: 
https://github.com/rustdesk/rustdesk/releases/tag/1.3.0

You may need to reboot to see Rustdesk removed from your task tray and not running.
You can then start and close it the usual way and it should work fine on demand.

If you need to upgrade to a newer version, you may need to uninstall this version first.
You may then need to repeat to remove it from systemd startup.

-Marc

**Comment by delinuxco** [comment]
@tethragon Sounds like you would be better off using the rustdesk.AppImage package. The AppImage will not stay runnign when closed.
